nixos/modules/services/admin/nomachine.nix

```diff
@@ -0,0 +1,310 @@
+      (attrsOf (nullOr (oneOf [str int bool attrs])))
+      // {
+        description = "settings option";
+      };
+
+    format = value:
+      lib.generators.toKeyValue {
+        mkKeyValue = k: v:
+          if lib.isAttrs v
+          then "Section \"${k}\"\n" + (format v) + "EndSection"
+          else if lib.isBool v
+          then
+            lib.generators.mkKeyValueDefault {} " " k (
+              if v == true
+              then 1
+              else 0
+            )
+          else if lib.isString v
+          then
+            lib.generators.mkKeyValueDefault {} " " k (
+              if k == "Name"
+              then "\"${v}\""
+              else v
+            )
+          else lib.generators.mkKeyValueDefault {} " " k v;
+      }
+      value;
+
+    generate = name: value:
+      pkgs.writeText name (
+        format value
+      );
+  };
+in {
+  options.services.nxserver = with lib; {
+    enable = mkEnableOption "the NoMachine remote desktop server";
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to open the configured port in the firewall.
+      '';
+    };
+
+    package = mkOption {
+      description = ''
+        Package containing the nxserver package and configuration files.
+      '';
+      type = types.package;
+      default = pkgs.nomachine;
+      defaultText = literalExpression "pkgs.nomachine";
+    };
+
+    serverSettings = lib.mkOption {
+      description = ''
+        Settings for the NoMachine nxserver instance.
+      '';
+
+      default = {};
+      example = {
+        EnableDebug = true;
+        SessionLogLevel = 6;
+      };
+
+      type = lib.types.submodule {
+        freeformType = settingsFmt.type;
+
+        options.EnableDebug = lib.mkEnableOption "debug output";
+        options.SessionLogLevel = lib.mkOption {
+          type = lib.types.int;
+          default = 6;
+          example = 6;
+          description = "Debug output level";
+        };
+      };
+    };
+
+    nodeSettings = lib.mkOption {
+      description = ''
+        Settings for the NoMachine nxnode instance.
+      '';
+
+      default = {};
+      example = {
+        EnableDebug = true;
+        SessionLogLevel = 6;
+      };
+
+      type = lib.types.submodule {
+        freeformType = settingsFmt.type;
+
+        options.EnableDebug = lib.mkEnableOption "debug output";
+        options.SessionLogLevel = lib.mkOption {
+          type = lib.types.int;
+          default = 6;
+          example = 6;
+          description = "Debug output level";
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    users.users = {
+      "nx" = {
+        group = "nx";
+        isSystemUser = true;
+        shell = "${cfg.package}/NX/etc/nxserver";
+        home = "/var/lib/nxserver/nx";
+      };
+    };
+
+    users.groups = {
+      "nx" = {};
+    };
+
+    services.nxserver = {
+      serverSettings = {
```

Comment:
I get a bunch of firewall errors in the logs where it's trying to make changes to `iptables`, which fails. 

```
104940 104940 2022-04-16 14:46:25 770.929 NXSERVER WARNING! Process '/nix/store/1vm2kik66jxjdmr5li9avyj64k5c7r0g-nomachine-7.9.2/NX/bin/nxexec nxfwget.sh ip6tables INPUT' with pid '104991/104991' finished with exit code 1 after 0,010 seconds.
104940 104940 2022-04-16 14:46:25 771.166 NXSERVER WARNING! Cannot check ip6tables status. Exit value '1'. Please check the 'nxerror.log' file for possible issues.                                                    104940 104940 2022-04-16 14:46:25 783.256 NXSERVER WARNING! Process '/nix/store/1vm2kik66jxjdmr5li9avyj64k5c7r0g-nomachine-7.9.2/NX/bin/nxexec nxfwget.sh iptables INPUT' with pid '104994/104994' finished with exit code 1 after 0,011 seconds.
104940 104940 2022-04-16 14:46:25 783.429 NXSERVER WARNING! Cannot check iptables status. Exit value '1'. Please check the 'nxerror.log' file for possible issues.                                                     104940 104940 2022-04-16 14:46:25 795.454 NXSERVER WARNING! Process '/nix/store/1vm2kik66jxjdmr5li9avyj64k5c7r0g-nomachine-7.9.2/NX/bin/nxexec nxfwadd.sh iptables 4000 tcp 1' with pid '104997/104997' finished with exit code 1 after 0,010 seconds.
104940 104940 2022-04-16 14:46:25 795.633 NXSERVER WARNING! Cannot add firewall rule for '4000' with 'iptables' command. Exit value '1'. Please check the nxerror.log file for possible issues.
104940 104940 2022-04-16 14:46:25 807.477 NXSERVER WARNING! Process '/nix/store/1vm2kik66jxjdmr5li9avyj64k5c7r0g-nomachine-7.9.2/NX/bin/nxexec nxfwadd.sh ip6tables 4000 tcp 1' with pid '105000/105000' finished with exit code 1 after 0,010 seconds.
104940 104940 2022-04-16 14:46:25 807.650 NXSERVER WARNING! Cannot add firewall rule for '4000' with 'ip6tables' command. Exit value '1'. Please check the nxerror.log file for possible issues.
104940 104940 2022-04-16 14:46:25 819.608 NXSERVER WARNING! Process '/nix/store/1vm2kik66jxjdmr5li9avyj64k5c7r0g-nomachine-7.9.2/NX/bin/nxexec nxfwadd.sh iptables 5353 udp 1' with pid '105003/105003' finished with exit code 1 after 0,010 seconds.
104940 104940 2022-04-16 14:46:25 819.783 NXSERVER WARNING! Cannot add firewall rule for '5353' with 'iptables' command. Exit value '1'. Please check the nxerror.log file for possible issues.
104940 104940 2022-04-16 14:46:25 831.840 NXSERVER WARNING! Process '/nix/store/1vm2kik66jxjdmr5li9avyj64k5c7r0g-nomachine-7.9.2/NX/bin/nxexec nxfwadd.sh ip6tables 5353 udp 1' with pid '105006/105006' finished with exit code 1 after 0,010 seconds.
104940 104940 2022-04-16 14:46:25 832.075 NXSERVER WARNING! Cannot add firewall rule for '5353' with 'ip6tables' command. Exit value '1'. Please check the nxerror.log file for possible issues.
```

Guessing we should add `EnableFirewallConfiguration = false;`.

Reply:
Makes sense, I added it to the `serverSettings`.